### PR TITLE
Add warnings when in an infinite loop

### PR DESCRIPTION
**Pull request by jcoyne**
## Why was this change made?

To point people who are debugging to the right place in the code.

### CODE REVIEW

File: app/services/administrative_tags.rb [hunk 1]
```diff
@@ -127,7 +129,9 @@ def update(current:, new:)
     # If two threads are creating the same tag, one will get an exception.
     # We must catch this outside the transaction block, because once a constraint
     # is violated, PG will permit no more statements in that transaction.
+    # When we go to rails 6 we can replace find_or_create_by with create_or_find_by: https://sikac.hu/use-create-or-find-by-to-avoid-race-condition-in-rails-6-0-f44fca97d16b
 
+    Rails.logger.warn("Possible race condition updating tags: #{tags}.  This should only happen one time, otherwise this might be an error")
```
Comment (jmartin-sul) [inline app/services/administrative_tags.rb:134]: apologies if i'm missing something, but is there actually a var called `tags` in scope here?

Comment (jmartin-sul) [reply app/services/administrative_tags.rb:134]: i see it as a param to the method in the other added logging statement

**Review by jmartin-sul — CHANGES_REQUESTED** [review]
i think one of the logging statements might reference an unavailable variable

File: app/services/administrative_tags.rb (hunk 1 above)
Comment (jcoyne) [reply app/services/administrative_tags.rb:134]: Good catch, should be `new`

**Comment by jcoyne** [comment]
Superseded by #3048

**Pull request closed without merging**